File: src/main/java/com/ptah/entity/contract/Contract.java
package com.ptah.entity.contract;

import com.ptah.common.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
public class Contract extends BaseEntity {

    @ManyToOne
    private Participant partyA;
    @ManyToOne
    private Participant partyB;
    private BigDecimal amount;
    @Enumerated(value = EnumType.STRING)
    private ContractType type;
}

File: src/main/kotlin/com/ptah/entity/contract/Contract.kt
package com.ptah.entity.contract

import com.ptah.common.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.ManyToOne
import java.math.BigDecimal

@Entity
class Contract : BaseEntity() {
    @ManyToOne
    val partyA: Participant? = null

    @ManyToOne
    val partyB: Participant? = null
    val amount: BigDecimal? = null

    @Enumerated(value = EnumType.STRING)
    val type: ContractType? = null
}
